### ISSUE TITLE
WARNING "Failed to import 1 files from ..." caused by empty "segments"  block in NZB

### ISSUE THREAD

**Issue by sanderjo** [sabnzbd/sabnzbd#2531]
People are reporting problems with "Failed to import 1 files from ...". For example https://www.reddit.com/r/SABnzbd/comments/12nmngg/getting_failed_to_import_1_files_from_filenamenzb/ and https://forums.sabnzbd.org/viewtopic.php?t=26199 . 

Cause: empty segments block in the NZB. Easy to reproduce:
```
	<file poster="blablamannetje &lt;blabla@example.com&gt;" date="1681636714" subject="reftestnzb 100 and 1 byte 1b77f77eb049 [1/2] - &quot;100byte.bin&quot; yEnc (1/1) 100">
		<groups>
			<group>alt.binaries.test</group>
		</groups>
		<segments>
		</segments>
	</file>
```

https://www.w3schools.com/xml/xml_elements.asp#:~:text=(category%3D%22children%22).-,Empty%20XML%20Elements,-An%20element%20with seems to suggest that an empty element / block is allowed by XML. I'm not sure if it's allowed by the NZB spec: https://sabnzbd.org/wiki/extra/nzb-spec is unclear to me. If SAB would allow it ... then create a 0 byte file ... ? Or just ignore it and do not yield a WARNING at all?

I tried the other way around: 100 byte file and 0 byte file, and let nyuu post it. 
```
-rw-rw-r-- 1 sander sander    0 apr 16 11:27 0byte.bin
-rw-rw-r-- 1 sander sander  100 apr 16 11:27 100byte.bin
```

The resulting NZB has no 0 byte in it, and nyuu tells so:

`[WARN] Skipping empty file: 0byte.bin`
So maybe @animetosho can tell why nyuu skips that 0 byte file? Not allowed?

**Comment by sanderjo** [comment]
TL;DR:

What if we change https://github.com/sabnzbd/sabnzbd/blob/c2431440093e2b48ddc28c46115cbcbd55ab49a0/sabnzbd/nzbparser.py#L493
to a logging.info (not a warning)?

**Comment by Safihre** [comment]
Sigh, why would they do such a stupid thing even if it is allowed?
We're not changing it to Info, but we can skip empty files in another way.

**Comment by animetosho** [comment]
> So maybe @animetosho can tell why nyuu skips that 0 byte file? Not allowed?

I don't recall any of the specs disallowing it, though I also don't recall them clarifying how it would work. I mostly figured that it's probably not something you'd want to do (and could be a sign of someone uploading a file that hasn't been written yet).

Maybe I can allow it, though it's not clear how to do it - I see two main approaches:

1. Don't upload anything and list the file in the NZB without any segments
2. Upload an article with an empty yEnc'd file; the file in the NZB will have one segment, despite being 0 bytes

The latter probably makes more sense, as it allows the downloader to obtain the filename without demangling the subject in the NZB, and follows the notion of the NZB representing what an indexer could produce. But I don't like the lack of clarity, and as such, don't know how well it would be handled/received.

(relying on PAR2 handle it might be an alternative, if it were not for [par2cmdline not handling it well either](https://github.com/Parchive/par2cmdline/issues/130#issuecomment-489492585))

Aside: the NZB specification is terrible.

**Comment by puzzledsab** [comment]
I think we should continue to give a clear warning. The nzbs are genuinely broken because having a 0 byte par2 file makes no sense and the indexer or uploader should fix it and stop serving them in this state. If it was a data file without par2 it could result in a broken download with no message about anything being wrong. SAB could be more helpful about what's wrong, though.

**Comment by Safihre** [comment]
> I think we should continue to give a clear warning. The nzbs are genuinely broken because having a 0 byte par2 file makes no sense and the indexer or uploader should fix it and stop serving them in this state. If it was a data file without par2 it could result in a broken download with no message about anything being wrong. SAB could be more helpful about what's wrong, though.

Agreed 👍

**Comment by sanderjo** [comment]
> SAB could be more helpful about what's wrong, though.

I propose adding "Error in NZB-file. Contact your Indexer.", to both messages.

logging.info("Error importing %s, skipping", file_name)
logging.warning(T("Failed to import %s files from %s"), skipped_files, nzo.filename)
